### DISCUSSION TITLE
Relative checkstyle.config.location possible

### DISCUSSION THREAD

**Discussion by Auto81** [checkstyle/checkstyle#14093] [Q&A] [answered]
Hi All,

I have a setup where there are multiple maven based projects stored within the same repo. Is there a way for me to reference a common/shared checkStyle file at the root level of the repo from each of the projects, using a relative path for the <checkstyle.config.location> variable in each project? Or should I have copies in each project?

If I provide the absolute path to the checkStyle file in the Repo `/Users/Repo/checkStyle.xml` in each project it works, and raises errors/violations as expected.

However if i switch it to a relative path in each project `${basedir}/../../checkStyle.xml` it seems to find the checkStyle file but doesn't raise any errors/violations. If I provide a bad relative path `${basedir}/../checkStyle.xml` it will fail during checkstyle saying it can't find the file.

```
Repo
->checkStyle.xml
-->Project1
-->Project2
```
-------
```
    <properties>
        <maven.compiler.plugin.version>3.11.0</maven.compiler.plugin.version>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <maven.compiler.target>21</maven.compiler.target>
        <maven.compiler.source>21</maven.compiler.source>
        <maven.compiler.version>21</maven.compiler.version>
        <java.version>21</java.version>
        <checkstyle.version>10.12.5</checkstyle.version>
        <checkstyle.plugin.version>3.3.1</checkstyle.plugin.version>

        <checkstyle.config.location>
            <!--/Users/Repo/checkStyle.xml-->
            <!--${basedir}/../../checkStyle.xml-->
        </checkstyle.config.location>
    </properties>
```

**Comment by rnveach** [comment, marked as answer]
This is a maven-checkstyle issue, and not a checkstyle issue. They are a different plugin.

I don't think they ever fully resolved this and I believe you can find ways around this googling some. I don't remember any specifics.

**Reply by Auto81** [reply]
Ah wrong place, thanks!
